tests/core/arch-arm64/stalker-arm64.c

```diff
@@ -0,0 +1,660 @@
+    g_cond_wait (&ctx->cond, &ctx->mutex);
+
+  g_mutex_unlock (&ctx->mutex);
+
+  return NULL;
+}
+
+STALKER_TESTCASE (follow_thread)
+{
+  StalkerVictimContext ctx;
+  GumThreadId thread_id;
+  GThread * thread;
+
+  ctx.state = STALKER_VICTIM_CREATED;
+  g_mutex_init (&ctx.mutex);
+  g_cond_init (&ctx.cond);
+
+  thread = g_thread_new ("stalker-test-victim", stalker_victim, &ctx);
+
+  /* 1: Wait for victim to tell us it's ready, giving its thread id */
+  g_mutex_lock (&ctx.mutex);
+  while (ctx.state != STALKER_VICTIM_READY_FOR_FOLLOW)
+    g_cond_wait (&ctx.cond, &ctx.mutex);
+  thread_id = ctx.thread_id;
+  g_mutex_unlock (&ctx.mutex);
+
+  /* 4: Follow and notify victim about it */
+  fixture->sink->mask = (GumEventType) (GUM_EXEC | GUM_CALL | GUM_RET);
+  gum_stalker_follow (fixture->stalker, thread_id,
+      GUM_EVENT_SINK (fixture->sink));
+  g_mutex_lock (&ctx.mutex);
+  ctx.state = STALKER_VICTIM_IS_FOLLOWED;
+  g_cond_signal (&ctx.cond);
+  g_mutex_unlock (&ctx.mutex);
+
+  /* 5: Wait for victim to tell us to unfollow */
+  g_mutex_lock (&ctx.mutex);
+  while (ctx.state != STALKER_VICTIM_READY_FOR_UNFOLLOW)
+    g_cond_wait (&ctx.cond, &ctx.mutex);
+  g_mutex_unlock (&ctx.mutex);
+
+  g_assert_cmpuint (fixture->sink->events->len, >, 0);
+
+  /* 8: Unfollow and notify victim about it */
+  gum_stalker_unfollow (fixture->stalker, thread_id);
+  g_mutex_lock (&ctx.mutex);
+  ctx.state = STALKER_VICTIM_IS_UNFOLLOWED;
+  g_cond_signal (&ctx.cond);
+  g_mutex_unlock (&ctx.mutex);
+
+  /* 9: Wait for victim to tell us it's ready for us to reset the sink */
+  g_mutex_lock (&ctx.mutex);
+  while (ctx.state != STALKER_VICTIM_READY_FOR_SHUTDOWN)
+    g_cond_wait (&ctx.cond, &ctx.mutex);
+  g_mutex_unlock (&ctx.mutex);
+
+  gum_fake_event_sink_reset (fixture->sink);
+
+  /* 12: Tell victim to it' */
+  g_mutex_lock (&ctx.mutex);
+  ctx.state = STALKER_VICTIM_IS_SHUTDOWN;
+  g_cond_signal (&ctx.cond);
+  g_mutex_unlock (&ctx.mutex);
+
+  g_thread_join (thread);
+
+  g_assert_cmpuint (fixture->sink->events->len, ==, 0);
+
+  g_mutex_clear (&ctx.mutex);
+  g_cond_clear (&ctx.cond);
+}
+
+STALKER_TESTCASE (heap_api)
+{
+  gpointer p;
+
+  fixture->sink->mask = (GumEventType) (GUM_EXEC | GUM_CALL | GUM_RET);
+
+  gum_stalker_follow_me (fixture->stalker, GUM_EVENT_SINK (fixture->sink));
+  p = malloc (1);
+  free (p);
+  gum_stalker_unfollow_me (fixture->stalker);
+
+  g_assert_cmpuint (fixture->sink->events->len, >, 0);
+}
+
+typedef void (* ClobberFunc) (GumCpuContext * ctx);
+
+STALKER_TESTCASE (no_register_clobber)
+{
+  guint8 * code;
+  GumArm64Writer cw;
+  const gchar * my_func_lbl = "my_func";
+  const gchar * my_beach_lbl = "my_beach";
+  const gchar * my_ken_lbl = "my_ken";
+  gint i;
+  gint offset;
+  ClobberFunc func;
+  GumCpuContext ctx;
+
+  code = gum_alloc_n_pages (1, GUM_PAGE_RWX);
+  gum_arm64_writer_init (&cw, code);
+
+  gum_arm64_writer_put_push_all_x_registers (&cw);  /* 16 push of 16 */
```

Comment:
Let's kill this comment or fix it by removing the extra space before it, and say push*es*, e.g. `16 pushes of 16 bytes each`.